agent-control/src/sub_agent/on_host/supervisor.rs

```diff
@@ -60,18 +60,27 @@ impl SupervisorStarter for NotStartedSupervisorOnHost {
     ) -> Result<Self::SupervisorStopper, SupervisorStarterError> {
         let ctx = self.ctx.clone();
 
-        let thread_contexts = vec![
+        let executable_thread_contexts =
+            self.executables.clone().into_iter().map(|e| {
+                self.start_process_thread(sub_agent_internal_publisher.clone(), e.clone())
+            });
```

Comment:
I _think_ the `start_process_thread` would be fine if it accepted a `&ExecutableData` instead of an owned value, since it is clone already inside before spawning the thread (even several of its fields before cloning the whole thing).

I also think this iterator could avoid both being cloned and instead use only `iter`, `map` over the references instead, and and pass the closure arg directly. Did you try this?

Reply:
Modified in last commit.